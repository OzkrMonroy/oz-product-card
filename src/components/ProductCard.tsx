import React, { createContext } from 'react';
import styles from '../styles/styles.module.css';
import {
  ProductCardProps,
  ProductContextProps,
} from '../interfaces/productInterfaces';
import { useProduct } from '../hooks/useProduct';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export const ProductCard = ({
  product,
  children,
  className,
  style,
  onChange,
  count,
  initialValues,
}: ProductCardProps) => {
  const { counter, increaseBy, isMaxCountReached, reset } = useProduct({
    product,
    onChange,
    initialCounter: count,
    initialValues,
  });

  return (
    <Provider
      value={{
        product,
        counter,
        increaseBy,
        maxCount: initialValues?.maxCount,
      }}
    >
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          reset,
          maxCount: initialValues?.maxCount,
          product,
          increaseBy,
        })}
        {/* <ProductImage img={img} name={name} />
      <ProductTitle name={name} />
      <ProductsButtons counter={counter} increaseBy={increaseBy}/> */}
      </div>
    </Provider>
  );
};
