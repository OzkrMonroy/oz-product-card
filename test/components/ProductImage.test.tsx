import * as React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../mocks/products';

describe('Product Image component', () => {
  test('should render the image that is passed by props', () => {
    const wrapper = renderer.create(
      <ProductImage
        img={'https://beautifulurl.com/image.jpg'}
        name="Custom image"
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('should display the product image when nothing is sended', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
