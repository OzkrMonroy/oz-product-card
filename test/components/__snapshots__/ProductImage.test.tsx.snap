// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image component should display the product image when nothing is sended 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="This is an example - With image"
    className="productImg undefined"
    src="https://awesomeurl.com/product.jpg"
  />
</div>
`;

exports[`Product Image component should render the image that is passed by props 1`] = `
<img
  alt="Custom image"
  className="productImg undefined"
  src="https://beautifulurl.com/image.jpg"
/>
`;
